Analytics-CSharp/Segment/Analytics/Analytics.cs

```diff
@@ -229,29 +196,32 @@ public void Reset()
 
             return returnSettings;
         }
-        
+
         #endregion
-        
-        
+
+
 
         #region Startup
 
         private void Startup(HTTPClient httpClient = null)
         {
             Add(new StartupQueue());
             Add(new ContextPlugin());
+            Add(new UserInfoPlugin());
 
             // use Wait() for this coroutine to force completion,
             // since Store must be setup before any event call happened.
             // Note: Task.Wait() forces internal async methods to run in a synchronized way,
             // we should avoid of doing it whenever possible.
             analyticsScope.Launch(analyticsDispatcher, async () =>
             {
+                // load memory with initial value
+                userInfo = UserInfo.DefaultState(configuration, storage);
                 await store.Provide(UserInfo.DefaultState(configuration, storage));
```

Comment:
we can provide the cache copy instead of providing another default state, since this leads to another read operation